hypershift-operator/controllers/hostedcluster/internal/platform/powervs/ccm.go

```diff
@@ -0,0 +1,294 @@
+		Version:                  "1.1.0",
+		ConfigFile:               fmt.Sprintf("%s/%s", kubeConfigFileBasePath, kubeConfigKeyInSecret),
+		ClusterDefaultProvider:   "g2",
+		AccountID:                accountID,
+		ClusterID:                ccm.hcluster.Name,
+		G2workerServiceAccountID: accountID,
+		G2Credentials:            fmt.Sprintf("%s/%s", secretMountPath, "ibmcloud_api_key"),
+		G2ResourceGroupName:      ccm.hcluster.Spec.Platform.PowerVS.ResourceGroup,
+		G2VpcSubnetNames:         ccm.hcluster.Spec.Platform.PowerVS.VPC.Subnet,
+		G2VpcName:                ccm.hcluster.Spec.Platform.PowerVS.VPC.Name,
+		Region:                   ccm.hcluster.Spec.Platform.PowerVS.VPC.Region,
+		PowerVSCloudInstanceID:   ccm.hcluster.Spec.Platform.PowerVS.ServiceInstanceID,
+		PowerVSRegion:            ccm.hcluster.Spec.Platform.PowerVS.Region,
+		PowerVSZone:              ccm.hcluster.Spec.Platform.PowerVS.Zone,
+	}
+
+	templateData := `
+[global]
+version = {{.Version}}
+[kubernetes]
+config-file = {{.ConfigFile}}
+[provider]
+cluster-default-provider = {{.ClusterDefaultProvider}}
+accountID = {{.AccountID}}
+clusterID = {{.ClusterID}}
+g2workerServiceAccountID = {{.G2workerServiceAccountID}}
+g2Credentials = {{.G2Credentials}}
+g2ResourceGroupName = {{.G2ResourceGroupName}}
+g2VpcSubnetNames = {{.G2VpcSubnetNames}}
+g2VpcName = {{.G2VpcName}}
+region = {{.Region}}
+powerVSCloudInstanceID = {{.PowerVSCloudInstanceID}}
+powerVSRegion = {{.PowerVSRegion}}
+powerVSZone = {{.PowerVSZone}}`
+
+	configMapTemplate := template.New("configMap")
+	configMapTemplate.Parse(templateData)
+	configMapData := &bytes.Buffer{}
+	configMapTemplate.Execute(configMapData, configMap)
+
+	configMapObj := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configMapName,
+			Namespace: fmt.Sprintf("%s-%s", ccm.hcluster.Namespace, ccm.hcluster.Name),
+		},
+		Data: map[string]string{
+			configMapName: configMapData.String(),
+		},
+	}
+
+	_, err = ccm.createOrUpdate(ccm.ctx, ccm.c, configMapObj, func() error {
+		return nil
+	})
+
+	return
+}
+
+func (ccm setupCCM) setupDeployment() (err error) {
+	commandToExec := []string{
+		"/bin/ibm-cloud-controller-manager",
+		"--authentication-skip-lookup",
+		"--bind-address=$(POD_IP_ADDRESS)",
+		"--use-service-account-credentials=true",
+		"--configure-cloud-routes=false",
+		"--cloud-provider=ibm",
+		fmt.Sprintf("--cloud-config=%s/%s", configMapMountPath, configMapName),
+		"--profiling=false",
+		"--leader-elect=true",
+		"--leader-elect-lease-duration=137s",
+		"--leader-elect-renew-deadline=107s",
+		"--leader-elect-retry-period=26s",
+		"--tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_AES_128_GCM_SHA256,TLS_CHACHA20_POLY1305_SHA256,TLS_AES_256_GCM_SHA384",
+		fmt.Sprintf("--v=%d", 2),
```

Comment:
what's the default logging? if we aren't changing this then why using Sprintf here?

Reply:
default might be 1, after removing this could nt see few logs. But for now we can leave it to default I guess.